test/Tests/src/Network/Http_Spec.enso

```diff
@@ -239,4 +241,38 @@ spec =
             test_file = enso_project.data / "does_not_exist.txt"
             Data.post url_post (Request_Body.Binary test_file) . should_fail_with Request_Error
 
+    Test.group "Headers" <|
+        base_url_with_slash = if base_url.ends_with "/" then base_url else base_url + "/"
+        url_post = base_url_with_slash + "post"
+
+        Test.specify "Content-type in the body is respected" <|
+            response = Data.post url_post (Request_Body.Text '{"a": "asdf", "b": 123}' content_type="application/json")
+            response.at "headers" . at "Content-Type" . should_equal "application/json"
```

Comment:
What happens if two content types are specified?

Reply:
(I mean both of the two in the headers)

Reply:
Only one content type can be passed in the body, as this test is doing. If multiple content types are passed in the header list, they are left unchanged.

Reply:
I'll add a test for this.